tools/depcheck/pkg/cmd/trace.go

```diff
@@ -0,0 +1,251 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"path"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/gonum/graph"
+	"github.com/gonum/graph/encoding/dot"
+
+	depgraph "github.com/openshift/origin/tools/depcheck/pkg/graph"
+)
+
+var traceImportsExample = `# create a dependency graph
+%[1]s trace --entry=pkg/foo --root=pkg/bar
+
+# create a dependency graph and output in "dot" format
+%[1]s trace --entry=pkg/foo --output=dot
+`
+
+type TraceOptions struct {
+	GraphOptions *GraphOptions
+
+	OutputFormat string
+
+	Out    io.Writer
+	ErrOut io.Writer
+}
+
+type TraceFlags struct {
+	GraphFlags *GraphFlags
+
+	OutputFormat string
+}
+
+func (o *TraceFlags) ToOptions(out, errout io.Writer) (*TraceOptions, error) {
+	graphOpts, err := o.GraphFlags.ToOptions(out, errout)
+	if err != nil {
+		return nil, err
+	}
+
+	return &TraceOptions{
+		GraphOptions: graphOpts,
+		OutputFormat: o.OutputFormat,
+
+		Out:    out,
+		ErrOut: errout,
+	}, nil
+}
+
+func NewCmdTraceImports(parent string, out, errout io.Writer) *cobra.Command {
+	traceFlags := &TraceFlags{
+		GraphFlags: &GraphFlags{},
+	}
+
+	cmd := &cobra.Command{
+		Use:     "trace --entry=pkg/foo",
+		Short:   "Creates a dependency graph from a set of given entrypoint packages",
+		Long:    "Creates a dependency graph from a set of given entrypoint packages",
+		Example: fmt.Sprintf(traceImportsExample, parent),
+		RunE: func(c *cobra.Command, args []string) error {
+			opts, err := traceFlags.ToOptions(out, errout)
+			if err != nil {
+				return err
+			}
+
+			if err := opts.Complete(); err != nil {
+				return err
+			}
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			if err := opts.Run(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	traceFlags.GraphFlags.AddFlags(cmd)
+	cmd.Flags().StringVarP(&traceFlags.OutputFormat, "output", "o", "dot", "output generated dependency graph in specified format. One of: dot.")
+	return cmd
+}
+
+func (o *TraceOptions) Complete() error {
+	if len(o.GraphOptions.RepoRoot) == 0 {
+		return errors.New("the --root path of the repository to analyze must be provided")
+	}
+
+	o.GraphOptions.Entrypoints = ensureEntrypointPrefix(o.GraphOptions.Entrypoints, o.GraphOptions.RepoRoot)
```

Comment:
Even if you did need this.  Why wouldn't it be a `Complete` on your `GraphOptions`?